### PR TITLE
Added parameter to select Content-Type when accessing the Rundeck API

### LINKED ISSUE #7683: rundeck module_utils: Allow to set content-type for api_request
### Summary

[`api_request`](https://github.com/ansible-collections/community.general/blob/1b9d437be8d085b714f2c38208b30401e7dcf2d9/plugins/module_utils/rundeck.py#L31) should allow to optionally set a custom content-type as this is necessary for some of the Rundeck API endpoints. A good example is the endpoint to upload keys to the storage ([Link](https://docs.rundeck.com/docs/api/#upload-keys)).

It should be good enough to add a new parameter:

```
def api_request(module, endpoint, data=None, method="GET", content_type="application/json"):
```

### Issue Type

Feature Idea

### Component Name

rundeck_project

### Additional Information

-

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by monsdar**
##### SUMMARY
Added optional parameter to set the desired Content-Type

Fixes #7683

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
rundeck_project, rundeck_acl_policy etc

### CODE REVIEW

File: changelogs/fragments/7683-added-contenttype-parameter.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Added parameter to select Content-Type when accessing the Rundeck API
```
Comment (felixfontein) [inline changelogs/fragments/7683-added-contenttype-parameter.yml:2]:
```suggestion
  - rundeck module utils - allow to pass ``Content-Type`` to API requests (https://github.com/ansible-collections/community.general/pull/7684).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! How do you plan to use this new parameter in the existin rundeck modules?

**Comment by monsdar** [comment]
> Thanks for your contribution! How do you plan to use this new parameter in the existin rundeck modules?

I don't think the existing modules need this option yet, as they're fine with `Content-Type: application/json`. For future (or private) extensions this is needed, e.g. with modules adding keys to the key storage.

**Comment by monsdar** [comment]
Understood. I'll see about a module about adding keys to the storage, so it's not just added complexity anymore.

**Comment by russoz** [comment]
LGTM other than the comments by @felixfontein

**Comment by felixfontein** [comment]
Hmm, looks like I wanted to already merge this, but forgot... let's do that now :)

**Comment by felixfontein** [comment]
@monsdar thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**